cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/BulkGetRequest.java

```diff
@@ -39,6 +39,36 @@ public BulkGetRequest(String id, List<String> revs, List<String> atts_since) {
     }
 
     @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        BulkGetRequest that = (BulkGetRequest) o;
+
+        if (id != null ? !id.equals(that.id) : that.id != null) {
+            return false;
+        }
+        if (revs != null ? !revs.equals(that.revs) : that.revs != null) {
+            return false;
+        }
+        return !(atts_since != null ? !atts_since.equals(that.atts_since) : that.atts_since !=
+                null);
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = id != null ? id.hashCode() : 0;
```

Comment:
Not essential, but I generally prefer to initialise `result` to a prime number before adding hash codes based on fields unless there's a good reason not to. I think that's the advice given (along with a good explanation) in _Effective Java_ by Josh Bloch. I think it's usually likely to give fewer hash collisions particularly when `id` is null.